File: Binary Trees/Traversals/09-Iterative Inorder Traversal of Binary Tree.java
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        if(root == null){
            return new ArrayList<>();
        }

        Stack<TreeNode> st = new Stack<>();
        List<Integer> traversal = new ArrayList<>();
        TreeNode current = root;
        
        while(!st.empty() || current != null){
            while(current != null){
                st.push(current);
                current = current.left;
            }

            current = st.pop();
            traversal.add(current.val);
            current = current.right;
        }

        return traversal;
    }
}

File: Binary Trees/Traversals/09-Iterative Inorder Traversal of Binary Tree.cpp
lass Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        if(root == nullptr){
            return {};
        }

        stack<TreeNode*> st;
        vector<int> traversal;
        TreeNode* current = root;
        
        while(!st.empty() || current != nullptr){
            while(current != nullptr){
                st.push(current);
                current = current -> left;
            }

            current = st.top();
            st.pop();

            traversal.push_back(current -> val);
            current = current -> right;
        }

        return traversal;
    }
};